src/riscv/linux/riscv-isa.c

```diff
@@ -0,0 +1,47 @@
+#include <string.h>
+#include <sys/auxv.h>
+
+#include <riscv/linux/api.h>
+
+/**
+ * arch/riscv/include/uapi/asm/hwcap.h
+ *
+ * This must be kept in sync with the upstream kernel header.
+ */
+#define COMPAT_HWCAP_ISA_I	(1 << ('I' - 'A'))
+#define COMPAT_HWCAP_ISA_M	(1 << ('M' - 'A'))
+#define COMPAT_HWCAP_ISA_A	(1 << ('A' - 'A'))
+#define COMPAT_HWCAP_ISA_F	(1 << ('F' - 'A'))
+#define COMPAT_HWCAP_ISA_D	(1 << ('D' - 'A'))
+#define COMPAT_HWCAP_ISA_C	(1 << ('C' - 'A'))
+#define COMPAT_HWCAP_ISA_V	(1 << ('V' - 'A'))
+
+void cpuinfo_riscv_linux_decode_isa_from_hwcap(
+		struct cpuinfo_riscv_isa isa[restrict static 1]) {
+	unsigned long hwcap = getauxval(AT_HWCAP);
+
+        // Erase any existing data prior to setting values.
+        memset(isa, 0, sizeof(struct cpuinfo_riscv_isa));
```

Comment:
Not needed, this function will be called only once and `isa` is zero-initialized by default